subtoaudio/subtoaudio.py

```diff
@@ -116,8 +122,21 @@ def convert_to_audio(self,
       print("Temporary folder:", temp_folder)
 
       for entry_data in data:
+        if cut_speaker_wav:
+          t1        = entry_data['start_time']
+          t2        = entry_data['end_time']
+          vocal     = fullvocal[t1:t2]
+          vocal.export(speaker_wav, format="wav")
+

```

Comment:
I'm not sure if this is your intention. Does this mean that every subtitle entry's voice will change according to the original voice track of the subtitle? So, is this a full voiceover?

Reply:
It kind of does, or it was my idea. The best would be to have a specific wav for each speaker, but it would require a lot of preparation.

I can describe my use case maybe it would be better to understand. My goal was to dub a movie because it only exists with subtitles, which my friend cannot read, so I wanted to generate a dub.

First, I extract the vocals by using https://github.com/Anjok07/ultimatevocalremovergui it takes some time, but I will have the "intrumental" track and vocal track.

I use the vocal track with sub-to-audio so that each subtitle uses its own portion of voice as speaker_wav. There are probably issues, like if one subtitle contains two speakers, but the idea was to have something work with limited effort.

At the end, I mix the instrumental, dubbed track and vocals with a lower volume because sometimes the vocals also extract normal sounds like crying, laughing...

Don't hesitate to ping me if something is not very clear in my explain.

Have a good day